build_utils/release_tox.ini

```diff
@@ -1,74 +1,49 @@
 [tox]
 # Use conda to set up the python environments to run in
-requires = tox-conda
+requires = tox>=4
 # The python environments to run the tests in
 envlist = pypi-py310-min,conda-py310-old-np,{pypi,conda}-{py310,py311,py312},pypisource-{py310,py312}
-# Skip the execution of setup.py as we do it with the correct version in commands_pre below
-skipsdist = True
+skip_install = True
 
 [testenv]
 changedir = tests_and_analysis/test
 test_command = python {toxinidir}/../tests_and_analysis/test/run_tests.py --report
 requirements_dir = {toxinidir}/../tests_and_analysis
+passenv =
+    EUPHONIC_VERSION
+    CC_LD
+setenv =
+    EXTRAS=[test,matplotlib,phonopy-reader,brille]
 
 # Test PyPI source distribution
 [testenv:pypisource-{py310,py312}]
-passenv = CC CC_LD LDFLAGS CPPFLAGS
-install_command = python -m pip install {opts} {packages}
-deps =
-    numpy
-    spglib  # Not interested here in whether source builds of spglib work
-    -r{[testenv]requirements_dir}/tox_requirements.txt
+passenv = CC, CC_LD, LDFLAGS, CPPFLAGS
 commands_pre =
     python -m pip install \
     --force-reinstall \
-    'euphonic[matplotlib,phonopy_reader,brille]=={env:EUPHONIC_VERSION}' \
+    'euphonic{env:EXTRAS}=={env:EUPHONIC_VERSION}' \
```

Comment:
Can we sub in-place using tox vars rather than environment here?

Reply:
Didn't seem to be allowed but I might have the syntax figured out now. One issue is that the name "extras" is reserved and we don't actually want a newline-separated list. So it has to be something else like "extras_str" 😞